packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages.tsx

```diff
@@ -259,14 +293,50 @@ function useCopyPastePages(): UseCopyPastePagesApi {
                     throw Error("Did not receive new uuid for page tree node");
                 }
 
-                // 1c. Create new document
+                // 1c. Copy all files used on page to target scope
+                const { data: filesOnPage } = await client.query<GQLGetAllFilesUsedOnPageQuery, GQLGetAllFilesUsedOnPageQueryVariables>({
+                    query: getAllFilesUsedOnPageQuery,
+                    variables: {
+                        pageTreeNodeId: node.id,
+                    },
+                });
+                const fileIds = filesOnPage.getAllFilesUsedOnPage.map((file) => file.id);
+
+                let newOutput: Record<string, unknown> | undefined;
+                if (fileIds.length > 0) {
+                    const { data: copiedFiles } = await client.mutate<GQLCopyFilesToScopeMutation, GQLCopyFilesToScopeMutationVariables>({
+                        mutation: copyFilesToScopeMutation,
+                        variables: { fileIds, targetScope: damScope, targetFolderId: inboxFolderIdForCopiedFiles },
+                        update: (cache, result) => {
+                            if (result.data && result.data.copyFilesToScope.numberNewlyCopiedFiles > 0) {
+                                cache.evict({ fieldName: "damItemsList" });
+                            }
+                        },
+                    });
+
+                    inboxFolderIdForCopiedFiles = copiedFiles?.copyFilesToScope.inboxFolderId;
+
+                    if (copiedFiles && node?.document != null && documentType.updateMutation && documentType.inputToOutput) {
+                        const output = documentType.inputToOutput(node.document, { idsMap });
+                        let stringifiedOutput = JSON.stringify(output);
+
+                        // TODO: implement a more graceful approach to replace the ids
+                        for (const mappedFile of copiedFiles.copyFilesToScope.mappedFiles) {
+                            stringifiedOutput = stringifiedOutput.replace(mappedFile.rootFile.id, mappedFile.copy.id);
+                        }
+
+                        newOutput = JSON.parse(stringifiedOutput);
```

Comment:
Recursively iterating the JSON and replacing any string without knowing anything about it would be brute force. But doing that on a JSON string is even more brute :)

Reply:
I replaced this with a more sophisticated check that navigates to the block props using the jsonPath from the block index. Then it searches all the props of the block and checks if they are equal to the old id. If so, the id is replaced.

That was the best solution I could come up with